app/services/flood_risk_engine/find_address_by_postcode.rb

```diff
@@ -0,0 +1,29 @@
+module FloodRiskEngine
+  class FindAddressByPostcode
+
+    attr_reader :address_result_set
+
+    def initialize(post_code)
+      Rails.logger.debug("FindAddressByPostcode supplied with params [#{post_code.inspect}]")
+      @post_code = post_code
+      @address_result_set = []
+    end
+
+    def search
+      @address_result_set = begin
+        EA::AddressLookup.find_by_postcode(post_code)
+      rescue => ex
+        Airbrake.notify(ex) if defined? Airbrake
+        Rails.logger.error "Address lookup service failed: #{ex}"
+        []
+      end
+
```

Comment:
I can see why the error handling is like this. Don't have a suggestion for this but in the rspec suite if a postcode test fails (eg VCR cassette url mismatch) the error does not show up and you have to dig in log/test.log to find the problem. Its the sort of thing a future developer could burn a couple of hours tracking down if they get a failing test like this. Maybe its poss to re-raise the error if `Rails.env.test?`and the error summary contains /VCR/?